telegram-bot-api/Client.cpp

```diff
@@ -2686,6 +2691,56 @@ class Client::JsonMessagesArray : public Jsonable {
   Client *client_;
 };
 
+class Client::JsonProxy : public Jsonable {
+ public:
+  explicit JsonProxy(object_ptr<td_api::proxy> &proxy) : proxy_(proxy) {
+  }
+
+  void store(JsonValueScope *scope) const {
+    auto object = scope->enter_object();
+    object("id", proxy_->id_);
+    object("last_used_date", proxy_->last_used_date_);
+    object("is_enabled", td::JsonBool(proxy_->is_enabled_));
+    object("server", proxy_->server_);
+    object("port", proxy_->port_);
+    switch (proxy_->type_->get_id()) {
+    case td_api::proxyTypeSocks5::ID:
+      {
+        auto ptype = static_cast<td_api::proxyTypeSocks5 *>(proxy_->type_.get());
+        object("type", "socks5");
+        object("username", ptype->username_);
+        object("password", ptype->password_);
+      }
+      break;
+    case td_api::proxyTypeMtproto::ID:
+      {
+        auto ptype = static_cast<td_api::proxyTypeMtproto *>(proxy_->type_.get());
+        object("type", "mtproto");
+        object("secret", ptype->secret_);
+      }
+      break;
+    };
```

Comment:
you're missing https://core.telegram.org/tdlib/docs/classtd_1_1td__api_1_1proxy_type_http.html